brave/brave-core
# Confirm NTP ad events

Resolves 

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Confirm "Estimated pending rewards" reflects the correct value after viewing a new tab page ad
- Confirm "Ad notifications received this month" reflects the correct value after viewing a new tab page ad
- Confirm ad notifications are shown in the 7-days ad history
- Confirm the following confirmations for new tab page ads are triggered:
    - Viewed (maximum 4 per hour and 20 per day)
    - Clicked
    - Landed
    - Flagged (Marked as inappropriate on 7-days ad history)
    - Upvoted (Thumbs up on 7-days ad history)
    - Downvoted (Thumbs down on 7-days ad history)
    - Conversion
- Confirm clicking a new tab page ad opens a new tab for the target url
- Confirm new tab page sponsored images appear if ads is disabled
- Confirm users are not rewarded for viewing new tab page ads if ads is disabled
- Confirm new tab page ads appear after upgrading from an older version of the browser

- Confirm "Estimated pending rewards" reflects the correct value after viewing an ad notification
- Confirm "Ad notifications received this month" reflects the correct value after viewing an ad notification
- Confirm new tab page ads are shown in the 7-days ad history
- Confirm the following confirmations for ad notifications are triggered:
    - Viewed
    - Clicked
    - Dismissed
    - Landed
    - Flagged (Marked as inappropriate on 7-days ad history)
    - Upvoted (Thumbs up on 7-days ad history)
    - Downvoted (Thumbs down on 7-days ad history)
    - Conversion
- Confirm clicking an ad notification opens a new tab for the target url
- Confirm a maximum 20 ad notifications are shown per day
- Confirm a maximum 5 ad notifications are shown per hour
- Confirm ad notifications appear after upgrading from an older version of the browser

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
